photon-lib/src/main/scala/com/linkedin/photon/ml/util/ClassUtils.scala

```diff
@@ -22,15 +22,15 @@ object ClassUtils {
   private val ANON_CLASS_MARKER = "$anon$"
 
   /**
-   * Method for detecting anonymous classes (the isAnonymousClass method is broken in Scala).
+   * Method for detecting anonymous classes (the isAnonymousClass method is broken in Scala as late as 2.11).
    *
    * @param clazz A class
    * @return True if the given class is anonymous, false otherwise.
    */
   def isAnonClass(clazz: Class[_]): Boolean = clazz.getName.contains(ANON_CLASS_MARKER)
```

Comment:
That we have this in production code suggests possible evil going on somewhere. Looks like it's used mainly when we infer GAME model types.